theano/tensor/nnet/corr.py

```diff
@@ -312,7 +330,7 @@ def c_code_helper(self, bottom, weights, top, direction, sub, height=None, width
         else {
           typenum = PyArray_TYPE(bottom);
         }
-        %(out)s = (PyArrayObject*)PyArray_EMPTY(4,
+        %(out)s = (PyArrayObject*)PyArray_ZEROS(4,
```

Comment:
I think we can still use EMPTY when there is only 1 thread. So it would be good to not slow that that case. Also, I think it need to be zero only for some of the direction. So try to minimize when this is set to zero. We could let it empty here and we set it to zero only when needed in the code bellow?